lib/puppet/file_system.rb

```diff
@@ -308,6 +308,14 @@ def self.size(path)
     @impl.size(assert_path(path))
   end
 
+  # @return [Time] the last-modified time of file
+  #
+  # @api public
+  #
+  def self.mtime(path)
+    @impl.mtime(assert_path(path))
+  end
```

Comment:
Do we default to adding wrappers to this module for all File methods as we need them, or only for ones that have OS-dependent behavior or custom processing that we need?

Reply:
Only when platform specific behavior diverges. Otherwise we share common implementations.

Reply:
@Iristyle Is there any platform specific behavior for mtim for a file needed here?  Should I adjust what I've done?